admin_manual/installation/php_configuration.rst

```diff
@@ -0,0 +1,132 @@
+
+* PHP module bz2 (recommended, required for extraction of apps)
+* PHP module intl (increases language translation performance and fixes sorting
+  of non-ASCII characters)
+* PHP module sodium (for Argon2 for password hashing. bcrypt is used as fallback, but if passwords were hashed with Argon2 already and the module is missing, your users can't log in.)
+
+Required for specific apps:
+
+* PHP module ldap (for LDAP integration)
+* PHP module smbclient  (SMB/CIFS integration, see
+  :doc:`../configuration_files/external_storage/smb`)
+* PHP module ftp (for FTP storage / external user authentication)
+* PHP module imap (for external user authentication)
+* PHP module bcmath (for passwordless login)
+* PHP module gmp (for passwordless login)
+
+Recommended for specific apps (*optional*):
+
+* PHP module gmp (for SFTP storage)
+* PHP module exif (for image rotation in pictures app)
+
+For enhanced server performance (*optional*) select one of the following
+memcaches:
+
+* PHP module apcu (>= 4.0.6)
+* PHP module memcached
+* PHP module redis (>= 2.2.6, required for Transactional File Locking)
+
+See :doc:`../configuration_server/caching_configuration` to learn how to select
+and configure a memcache.
+
+For preview generation (*optional*):
+
+* PHP module imagick
+* avconv or ffmpeg
+* OpenOffice or LibreOffice
+
+.. note::
+   If the preview generation of PDF files fails with a "not authorized" error message, you must adjust the imagick policy file.
+   See https://cromwell-intl.com/open-source/pdf-not-authorized.html
+
+For command line processing (*optional*):
+
+* PHP module pcntl (enables command interruption by pressing ``ctrl-c``)
+
+.. note::
+   You also need to ensure that ``pcntl_signal`` and ``pcntl_signal_dispatch`` are not disabled
+   in your php.ini by the ``disable_functions`` option.
+
+For command line updater (*optional*):
+
+* PHP module phar (upgrades Nextcloud by running ``sudo -u www-data php /var/www/nextcloud/updater/updater.phar``)
+
+ini values
+----------
+
+The following ini settings should be adapted if needed for Nextcloud:
+
+* ``apc.enable_cli``: See :doc:`../configuration_server/caching_configuration`
+* ``disable_functions``: Avoid disabling functions unless you know exactly what you are doing
+* ``max_execution_time``: See :doc:`../configuration_files/big_file_upload_configuration`
+* ``memory_limit``: Should at least 512MB. See also :doc:`../configuration_files/big_file_upload_configuration`
+* ``opcache.enable`` and friends: See :doc:`../configuration_server/caching_configuration` and :doc:`server_tuning`
+* ``open_basedir``: See :doc:`harden_server`
+* ``upload_tmp_dir``: See :doc:`../configuration_files/big_file_upload_configuration`
+
+.. _php_ini_tips_label:
+
+php.ini configuration notes
+---------------------------
+
+Keep in mind that changes to ``php.ini`` may have to be configured on more than one
+ini file. This can be the case, for example, for the ``date.timezone`` setting.
```

Comment:
```suggestion
ini file. This can be the case, for example, for the ``date.timezone`` setting.
You can search for a parameter with the following command: ``grep -r date.timezone /etc/php``.
```